scripts/run-registry.ts

```diff
@@ -71,59 +71,88 @@ const registryUrl = (command: string, url?: string) =>
 const registriesUrl = (yarnUrl?: string, npmUrl?: string) =>
   Promise.all([registryUrl('/usr/local/bin/yarn', yarnUrl), registryUrl('npm', npmUrl || yarnUrl)]);
 
+const cleanYarnConfig = () => {
+  return new Promise<string>((res, rej) => {
+    const commands = [
+      // FIXME: Use `yarn config unset` as soon as we will be using Yarn 3 in the monorepo
+      // `yarn config unset npmRegistryServer`,
+      // `yarn config unset npmAuthIdent`
+      `git restore .yarnrc.yml`,
+    ];
+
+    exec(commands.join(' && '), (e) => {
+      if (e) {
+        rej(e);
+      } else {
+        logger.log(`🛬 Yarn config has been cleanup`);
+
+        res(undefined);
+      }
+    });
+  });
+};
+
 const applyRegistriesUrl = (
   yarnUrl: string,
   npmUrl: string,
   originalYarnUrl: string,
   originalNpmUrl: string
 ) => {
   logger.log(`↪️  changing system config`);
-  nodeCleanup(() => {
-    registriesUrl(originalYarnUrl, originalNpmUrl);
-
-    logger.log(dedent`
-      Your registry config has been restored from:
-      npm: ${npmUrl} to ${originalNpmUrl} 
-      yarn: ${yarnUrl} to ${originalYarnUrl} 
-    `);
+  nodeCleanup((_exitCode, signal) => {
+    cleanYarnConfig()
+      .then(() => registriesUrl(originalYarnUrl, originalNpmUrl))
+      .then(() => {
+        logger.log(dedent`
+            Your registry config has been restored from:
+            npm: ${npmUrl} to ${originalNpmUrl} 
+            yarn: ${yarnUrl} to ${originalYarnUrl} 
+          `);
+      })
+      .then(() => process.kill(process.pid, signal));
+
+    nodeCleanup.uninstall();
+    return false;
   });
 
   return registriesUrl(yarnUrl, npmUrl);
 };
 
-const currentVersion = async () => {
-  const { version } = (await import('../lerna.json')).default;
-  return version;
-};
+const publish = (verdaccioUrl: string) => {
+  logger.log(`🛫 Publishing packages with a concurrency of ${maxConcurrentTasks}`);
 
-const publish = (packages: { name: string; location: string }[], url: string) => {
-  logger.log(`Publishing packages with a concurrency of ${maxConcurrentTasks}`);
-
-  const limit = pLimit(maxConcurrentTasks);
-  let i = 0;
-
-  return Promise.all(
-    packages.map(({ name, location }) =>
-      limit(
-        () =>
-          new Promise((res, rej) => {
-            logger.log(`🛫 publishing ${name} (${location})`);
-            const command = `cd ${location} && npm publish --registry ${url} --force --access restricted --ignore-scripts`;
-            exec(command, (e) => {
-              if (e) {
-                rej(e);
-              } else {
-                i += 1;
-                logger.log(`${i}/${packages.length} 🛬 successful publish of ${name}!`);
-                res(undefined);
-              }
-            });
-          })
-      )
-    )
-  );
+  return new Promise((res) => {
+    const commands = [
+      `yarn config set npmRegistryServer ${verdaccioUrl}`,
+      `yarn config set npmAuthIdent user:password`,
```

Comment:
You can set these in env variables to avoid mutating the config

```
YARN_NPM_REGISTRY_SERVER="${verdaccioUrl}"
YARN_NPM_AUTH_IDENT="user:password"
```